lib/std/fs/wasi.zig

```diff
@@ -22,6 +22,22 @@ pub const Preopens = struct {
         }
         return null;
     }
+
+    pub fn findDir(p: Preopens, full_path: []const u8, flags: std.fs.Dir.OpenDirOptions) std.fs.Dir.OpenError!std.fs.Dir {
+        var prefix: []const u8 = "";
+        var fd: usize = 0;
+        for (p.names) |preopen, i| {
+            if (i > 2 and wasiPathPrefixMatches(preopen, full_path)) {
+                if (preopen.len > prefix.len) {
+                    prefix = preopen;
+                    fd = i;
+                }
+            }
+        }
+        const d = std.fs.Dir{ .fd = @intCast(os.fd_t, fd) };
```

Comment:
This will implicitly open the first preopen if it didn't find any matching cases, or when there's less than 3 preopens. Is that expected behavior? If so, it would be nice to use a doc comment on top of `findDir` to tell the user what to expect.

Reply:
whoops, no!